orderbook/src/orderbook.rs

```diff
@@ -85,6 +91,48 @@ impl OrderBook {
         orders.retain(|_, order| has_future_valid_to(now_in_epoch_seconds, &order.order_creation));
     }
 
+    async fn remove_settled_orders(&self, settlement_contract: &GPv2Settlement) {
+        let uids_to_be_removed = self.get_uids_of_settled_orders(settlement_contract).await;
+        info!(
+            "removing following settled orders from orderbook: {:?}",
+            uids_to_be_removed
+        );
+        let mut orders = self.orders.write().await;
+        orders.retain(|uid, _| uids_to_be_removed.contains(uid));
+    }
+
+    async fn get_uids_of_settled_orders(
+        &self,
+        settlement_contract: &GPv2Settlement,
+    ) -> Vec<OrderUid> {
+        let orders = self.orders.read().await;
+        let uid_futures = orders.iter().map(|(uid, _)| async move {
+            self.has_uid_been_settled(*uid, settlement_contract).await
+        });
+        let uid_pairs: Vec<Option<OrderUid>> = join_all(uid_futures).await;
+        uid_pairs.iter().filter_map(|uid| *uid).collect()
+    }
+
+    async fn has_uid_been_settled(
+        &self,
+        uid: OrderUid,
+        settlement_contract: &GPv2Settlement,
+    ) -> Option<OrderUid> {
+        let filled_amount = settlement_contract
+            .filled_amount(uid.0.to_vec())
+            .call()
+            .await
+            .unwrap_or(U256::MAX);
```

Comment:
This unwrap_or means that if there is a connection error we will consider all orders as filled. Maybe on error it should consider as not filled?

Reply:
>Maybe on error it should consider as not filled?

My thought was: if the whole price calculation fails due to an already filled order, the damage is much bigger than not settling one particular order

Reply:
The problem is that if the node is unreachable this will trigger for all orders and remove all of them. And if the node is unreachable then the solver can't do it's work anyway.

Reply:
Yeah, I changed it. If there is an already filled order, then the computation is for nothing, but at least in the next run the orders can be settled again.